yaml.go

```diff
@@ -318,6 +316,17 @@ func convertToJSONableObject(yamlObj interface{}, jsonTarget *reflect.Value) (in
 	}
 }
 
+// getTarget gets the reflect.Value of the most preferential
+// struct field. If the field is a struct, this will recursively descend
+// into the definition.
+func getTarget(t reflect.Value, f *field) reflect.Value {
```

Comment:
Note this change is incomplete and causes a panic for "regular" nested structs.

```go
type Simple struct {
	A string `json:"a" yaml:"a"`
	B string `json:"b" yaml:"b"`
}

type Inner struct {
	A string `json:"a" yaml:"a"`
}

type WithInline struct {
	Inner Inner  `json:",inline" yaml:",inline"`
	B     string `json:"b" yaml:"b"`
}

type WithEmbed struct {
	Inner `json:",inline" yaml:",inline"`
	B     string `json:"b" yaml:"b"`
}
```

With the change from this PR in place, two of the 3 types work:

```go
s := Simple{}
err := yaml.Unmarshal([]byte("a: 10\nb: 11\n"), &s)
we := WithEmbed{}
err = yaml.Unmarshal([]byte("a: 10\nb: 11\n"), &we)
```

but a panic is caused by

```go
wi := WithInline{}
err = yaml.Unmarshal([]byte("a: 10\nb: 11\n"), &wi)
```